Engine/UI/Panel/PanelPopups.cpp

```diff
@@ -65,4 +72,81 @@ void PanelPopups::CreateScript()
 		if (ImGui::Button("Cancel", ImVec2(120, 0))) { ImGui::CloseCurrentPopup(); }
 		ImGui::EndPopup();
 	}
-}
\ No newline at end of file
+}
+
+void PanelPopups::SaveScene()
+{
+	if (save_scene_shown)
+	{
+		save_scene_shown = false;
+		ImGui::OpenPopup("Save Scene");
+	}
+
+	if (ImGui::BeginPopupModal("Save Scene", NULL, ImGuiWindowFlags_AlwaysAutoResize))
+	{
+		ImGui::Text("Scene name: \n");
+		ImGui::Separator();
+
+		static char str1[128] = "";
+		ImGui::InputTextWithHint("", "enter scene name here (XXXX.scene format)", str1, IM_ARRAYSIZE(str1));
+
+		if (ImGui::Button("Save", ImVec2(120, 0)))
+		{
+			std::string name = str1;
+			std::size_t found = name.find_last_of(".");
+			if (found == std::string::npos || found == 0)
+			{
+				APP_LOG_ERROR("Invalid name for scene, it should be of format XXXX.scene");
```

Comment:
maybe just putting "Invalid name for scene, introduce ".scene" extension", just a suggestion